### PR TITLE
UserDetails, Roles and services were added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

**Review by margarita-ignatieva — APPROVED** [review]

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 1]
```diff
@@ -33,8 +45,6 @@ protected void configure(HttpSecurity http) throws Exception {
                 .csrf().disable();
     }
 
-    @Bean
-    public PasswordEncoder getEncoder() {
-        return new BCryptPasswordEncoder();
-    }
 }
+
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: let's remove those empty lines

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .and()
+                .authorizeRequests()
+                .antMatchers(HttpMethod.POST, "/movies/**", "/cinema-halls/**",
+                        "/movie-sessions/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.POST, "/register/**",
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: register should be available for authenticated users only?

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl implements RoleDao {
+    private static final Logger logger = Logger.getLogger(ShoppingCartDaoImpl.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: shoppingCartDaoImpl ?

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            transaction = session.beginTransaction();
+            session.save(role);
+            transaction.commit();
+            logger.info("Role was created" + role);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Cant add role with id \"" + role.getId()
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: will role have id now?

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        }
+    }
+
+    @Override
+    public Role getRoleByName(String roleName) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Role> query = session.createQuery("from Role u where "
+                    + "u.roleName = :roleName", Role.class);
+            query.setParameter("roleName", roleName);
+            return query.uniqueResultOptional().get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: what value gives us calling uniqueResultOption() and get() right away? 
why not just uniqueResult?

File: src/main/java/com/dev/cinema/model/Role.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public void setRoleName(RoleName roleName) {
+        this.roleName = roleName;
+    }
+
+    public enum RoleName {
+        ADMIN, USER, MANAGER;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/Role.java]: do we use manager anywhere? 
if not let's add code when we need it

File: src/main/java/com/dev/cinema/security/impl/CustomUserDetailsServise.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+public class CustomUserDetailsServise implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsServise(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.getByEmail(email).orElse(null);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/security/impl/CustomUserDetailsServise.java]: use orElseThrow(our exception from else block) that will make code shorter

File: src/main/java/com/dev/cinema/service/impl/InjectData.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    public InjectData(UserService userService, RoleService roleService,
+                      ShoppingCartService shoppingCartService) {
+        this.userService = userService;
+        this.roleService = roleService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @PostConstruct
+    public void injector() {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/service/impl/InjectData.java]:
```suggestion
    public void injectDefaultData() {
```

File: src/main/java/com/dev/cinema/config/SecurityConfig.java (hunk 1 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/config/SecurityConfig.java]: done

File: src/main/java/com/dev/cinema/config/SecurityConfig.java (hunk 2 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/config/SecurityConfig.java]: done

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java (hunk 3 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: done

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java (hunk 4 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: done

File: src/main/java/com/dev/cinema/model/Role.java (hunk 6 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/model/Role.java]: Manager was deleted

File: src/main/java/com/dev/cinema/security/impl/CustomUserDetailsServise.java (hunk 7 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/security/impl/CustomUserDetailsServise.java]: done

File: src/main/java/com/dev/cinema/service/impl/InjectData.java (hunk 8 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/service/impl/InjectData.java]: done

**Comment by RomanDubovskyi** [comment]
what about this comment ?
image

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**